eventEmitter.js

```diff
@@ -0,0 +1,42 @@
+class EventEmitter {
+   listeners = {};
+     
+   on(eventName, fn) {
+      if(!this.listeners[eventName]) {
+         this.listeners[eventName] = [];
+      } 
+      this.listeners[eventName].push(fn) 
+   }
+        
+   off(eventName, fn) {
+      if (this.listeners[eventName]) {
+         this.listeners[eventName] = this.listeners[eventName].filter((handler) => handler !== fn);
```

Comment:
I would also check if there are no listeners for eventName and clean up the empty array.